src/Servers/IIS/test/IIS.Tests/Utilities/TestServer.cs

```diff
@@ -64,7 +66,7 @@ private TestServer(Action<IApplicationBuilder> appBuilder, ILoggerFactory logger
 
         public static async Task<TestServer> Create(Action<IApplicationBuilder> appBuilder, ILoggerFactory loggerFactory)
         {
-            await WebCoreLock.WaitAsync();
+            await WebCoreLock.WaitAsync(ServerTimeoutMilliseconds);
```

Comment:
What happens if a timeout is triggered? Does one IIS test fail, all IIS tests fail, etc?